beacon_chain/validators/beacon_validators.nim

```
@@ -1188,18 +1183,13 @@ proc proposeBlockAux(
             block_parent_root: forkyBlck.parent_root,
             proposer_index: forkyBlck.proposer_index,
             blob: blobs[i],
-            kzg_commitment: kzgs[i],
+            kzg_commitment: commitments[i],
             kzg_proof: proofs[i]
           )
           sidecars.add(sidecar)
         Opt.some(sidecars)
-      elif forkyBlck is phase0.BeaconBlock or
-           forkyBlck is altair.BeaconBlock or
-           forkyBlck is bellatrix.BeaconBlock or
-           forkyBlck is capella.BeaconBlock:
-        Opt.none(seq[BlobSidecar])
       else:
-        static: doAssert "Unknown BeaconBlock type"
```

Switching `static: doAssert` to `Opt.none` means that it won't trigger automatically anymore when the E-fork is added, but will incorrectly and silently use the pre-Deneb/sidecar version.

Better to find out about missing functionality at compile-time.

---

It will take the Deneb version, not the `none` version (`consensusFork >= ConsensusFork.Deneb`), and will compile-time fail if no longer compatible

---

The other one will compile-time break on Electra because `DenebBlockContents` requires a Deneb block, while the generated code will try to put an Electra block into it.

This instance here will be fine with Electra as well (using the blob logic from deneb)